vllm/distributed/parallel_state.py

```diff
@@ -357,6 +357,17 @@ def get_pipeline_model_parallel_prev_rank():
     return _PP_GLOBAL_RANKS[(rank_in_pipeline - 1) % world_size]
 
 
+def is_pipeline_model_parallel_first_rank() -> bool:
+    """Return True if the caller is the first rank in the pipeline"""
+    return get_pipeline_model_parallel_rank() == 0
+
+
+def is_pipeline_model_parallel_last_rank() -> bool:
+    """Return True if the caller is the last rank in the pipeline"""
+    return get_pipeline_model_parallel_rank(
+    ) == get_pipeline_model_parallel_world_size() - 1
```

Comment:
what's get_pipeline_model_parallel_world_size when tp2 pp2?

Reply:
It's gonna be 2